### PR TITLE
Fix AutoTrace_WriteToFile_ReadCorrectString

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Add Disabling logger before removing log file.
Improve functionality of AutoTrace_WriteToFile_ReadCorrectString

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
@VVeremjova @AByzhynar @okozlovlux @Anatoliy-Leshin @dtrunov 
Please review

File: src/components/utils/test/auto_trace_test.cc [hunk 1]
```diff
@@ -59,42 +60,74 @@ const std::string kFileName =
...
+  for (std::string line;
+       Compare<bool, EQ, ONE>(false, debug_found, trace_enter, trace_exit) &&
+           getline(file_log, line);) {
+    debug_found = debug_found
+                      ? debug_found
+                      : IsLogLineContains(line, debug_log_level, debug_message);
+    trace_enter = trace_enter
+                      ? trace_enter
+                      : IsLogLineContains(line, trace_log_level, enter_message);
+    trace_exit = trace_exit
```
Comment (VVeremjova) [inline src/components/utils/test/auto_trace_test.cc]: @LuxoftAKutsan It is possible situation that exit message won't be written. Maybe we shouldn't check exit message?

**Comment by okozlovlux** [comment]
@LuxoftAKutsan reviewed

File: src/components/utils/test/auto_trace_test.cc (hunk 1 above)
Comment (LuxoftAKutsan) [inline src/components/utils/test/auto_trace_test.cc]: @VVeremjova No. Exit message must be written

**Comment by VVeremjova** [comment]
Reviewed

**Comment by dtrunov** [comment]
Reviewed

**Comment by LuxoftAKutsan** [comment]
All test passed. Travis build failed due to very long build (The job exceeded the maximum time limit for jobs, and has been terminated.) Probably due to travs overload

**Comment by okozlovlux** [comment]
@LuxoftAKutsan reviewed

**Pull request merged**